perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -6058,6 +6087,43 @@ public void checkAttributeDependencies(PerunSession sess, RichAttribute richAttr
 	}
 
 	/**
+	 * Checks if the attributes represent empty values. If so, converts them into null.
+	 *
+	 * @param attributes attributes to be checked if are null
+	 */
+	public void convertEmptyAttrValueToNull(List<Attribute> attributes) {
+		attributes.forEach(this::convertEmptyAttrValueToNull);
+	}
+
+	/**
+	 * Checks if the attribute represents empty value. If so, converts it into null.
+	 *
+	 * @param attribute attribute to be checked if is null
+	 */
+	public void convertEmptyAttrValueToNull(Attribute attribute) {
+		if(attribute.getValue() == null) {
+			return;
+		}
+		if (attribute.getValue() instanceof String) {
+			if(((String)attribute.getValue()).matches("\\s*")) {
+				attribute.setValue(null);
+			}
+		} else if (attribute.getValue() instanceof Boolean) {
+			if(attribute.getValue().equals(Boolean.FALSE)) {
+				attribute.setValue(null);
+			}
+		} else if (attribute.getValue() instanceof ArrayList) {
+			if(((ArrayList)attribute.getValue()).isEmpty()) {
+				attribute.setValue(null);
+			}
+		} else if (attribute.getValue() instanceof HashMap) {
```

Comment:
We use ``LinkedHashMap`` as attribute value for a map. I would probably prefer solution, which checks against attribute type stored in a object (class names including large+className). What do you think @licehammer ? 
Now this method would ignore values of different java types, maybe use generic List/Map instead on ArrayList/HashMap ?